### PR TITLE
[WIP] SITES-632 Create RootSection to allow authors to create root level pages

### PR DESCRIPTION
**Pull request by maxious**
RootNode child nodes do appear as root level pages but without a section, you can't find them in editorial or give access to edit those pages etc. 
This PR attempts to add a special section type that can hold a RootNode and relax some of the UI/validations to let users easily manage root level pages.

### CODE REVIEW

File: app/controllers/admin/root_sections_controller.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+module Admin
+class TopicsController < Admin::ApplicationController
```
Comment (freshtonic) [inline app/controllers/admin/root_sections_controller.rb]: The class name does not match the file name.

File: app/models/general_content.rb [hunk 2]
```diff
@@ -3,7 +3,11 @@ class GeneralContent < Node
 
   validates_with SectionHeritageValidator
   validates_presence_of :section
-  validates :parent, ancestry_depth: { minimum: 2 }
+  validates :parent, ancestry_depth: { minimum: 2 }, if: :not_root_page?
+
+  def not_root_page?
+    not section.is_a? RootSection
```
Comment (freshtonic) [inline app/models/general_content.rb]: use `!` not `not`. Keyword versions of the ruby operators have different precedence rules and a generally frowned upon.

**Comment by freshtonic** [comment]
@maxious this is a clever approach to having non-section nodes at the root level but it builds upon (and increases) the "sections are special" tech debt. Can we not achieve the same thing by allowing GeneralContent nodes to be directly attached to the root, and update the UI to be aware of them?

**Comment by maxious** [comment]
GeneralContent nodes can be directly attached to the root but this was a neat way to get a UI. That being said, I'm starting to doubt that having more root level pages is the correct approach for this task.

**Comment by freshtonic** [comment]
The app implementation has a strong 1-1 correspondence to the IA, and I think this has bitten us a bit. This is why Sections were first-class domain entities instead of just Nodes. It's also why there are weird UI quirks in the editorial (doubled up on breadcrumbs for SectionHome, and "am I on the section or section home node" quirk).

I'd like to be able to defer this story until _after_ we have refactored the Section/SectionHome dichotomy. Maybe we can get some time tomorrow to discuss what needs to be done and estimate. It's long overdue IMHO.  @coderdan what do you think?

**Comment by maadonna** [comment]
Can you loop me in when you do talk about it (though if it's tomorrow I might be out with Thinkplace doing research). I'd like to understand why there might need to be pages (without being in a section) at the root. I'm not sure whether there is a need, or whether authors are asking for pages without thinking through whether they better belong in a section. I'm fairly confident we don't want distributed authors just creating pages at the top level.

**Comment by freshtonic** [comment]
@maadonna it's it allow the `/about-us` page to be edited. Currently it's hard coded. I expect that creating root level pages would require special/admin privileges.

**Comment by maadonna** [comment]
No worries. As I noted on the jira issue that it's being done in, I think all the footer pages should be in a 'About GOV.AU' section anyway, and then 'about gov.au' is the landing page for the section. But that's probably longer term than the immediate hack

**Pull request closed without merging**